hcl/fmt.go

```diff
@@ -151,3 +164,318 @@ func (f FormatResult) Formatted() string {
 const (
 	errFormatTree errors.Kind = "formatting tree"
 )
+
+func fmtBody(body *hclwrite.Body) {
+	logger := log.With().
+		Str("action", "hcl.fmtBody()").
+		Logger()
+
+	attrs := body.Attributes()
+	for name, attr := range attrs {
+		logger.Trace().
+			Str("name", name).
+			Msg("formatting attribute")
+		body.SetAttributeRaw(name, fmtAttrExpr(attr.Expr().BuildTokens(nil)))
+	}
+
+	blocks := body.Blocks()
+	for _, block := range blocks {
+		fmtBody(block.Body())
+	}
+}
+
+func fmtAttrExpr(tokens hclwrite.Tokens) hclwrite.Tokens {
+	logger := log.With().
+		Str("action", "hcl.fmtAttrExpr()").
+		Str("tokens", tokensStr(tokens)).
+		Logger()
+
+	logger.Trace().Msg("trimming newlines")
+
+	trimmed := trimNewlines(tokens)
+
+	logger = logger.With().
+		Str("trimmedTokens", tokensStr(trimmed)).
+		Logger()
+
+	// We are interested on lists, ignore the rest
+	if trimmed[0].Type == hclsyntax.TokenOBrack {
+
+		if isListComprehension(tokens) {
```

Comment:
```suggestion
		if isListComprehension(trimmed) {
```
this would be a bug but it's not because nothing is being trimmed anyway.

Reply:
what would be a bug ? checking if it is a list comprehension ? or something related to the trimming ? (already removed trimming here)

Reply:
removing the trimming fixed it, the issue was that isListComprehension() was checking the `tokens` and not the `trimmed` tokens. All good now.

Reply:
ahh indeed, and in that case it was wrong, but since no trimming ever happened it was working :laughing:. Thanks for catching that :100: